wdl2cwl/main_oop.py

```diff
@@ -66,76 +66,72 @@ def load_wdl_task(self, obj: WDL.Tree.Task) -> str:
-            )
-        )
+        requirements = self.get_cwl_requirements(obj.command, obj.runtime)
 
-        cat_tool = cwl.CommandLineTool(
+        cmd_tool = cwl.CommandLineTool(
             id=obj.name,
             inputs=cwl_inputs,
-            requirements=requirements,
             outputs=cwl_outputs,
+            requirements=requirements,
             cwlVersion="v1.2",
-            baseCommand=base_command,
+            baseCommand=["bash", "example.sh"],
         )
 
         yaml = YAML()
         yaml.default_flow_style = False
         yaml.indent = 4
         yaml.block_seq_indent = 2
-        result_stream = StringIO()
-        cwl_result = cat_tool.save()
+        cwl_result = cmd_tool.save()
         scalarstring.walk_tree(cwl_result)
+        result_stream = StringIO()
         yaml.dump(cwl_result, result_stream)
         yaml.dump(cwl_result, sys.stdout)
 
         return result_stream.getvalue()
 
-    def get_outdire_requirement(
+    def get_cwl_requirements(
+        self, command: WDL.Expr.String, runtime: WDL.Dict[str, WDL.Expr.Base]
+    ) -> List[cwl.ProcessRequirement]:
+        """Produce the CWL Requirements list."""
+        docker_requirement = self.get_cwl_docker_requirements(runtime["docker"])
+        cwl_command_str = self.get_cwl_command_requirements(command.parts)
+        cpu_requirement = (
+            self.get_cpu_requirement(runtime["cpu"]) if "cpu" in runtime else None
+        )
+        memory_requirement = (
+            self.get_memory_requirement(runtime["memory"])  # type: ignore
+            if "memory" in runtime
+            else None
+        )
+        outdir_requirement = (
+            self.get_outdir_requirement(runtime["disks"])  # type: ignore
+            if "disks" in runtime
+            else 1024
+        )
+        return [
+            docker_requirement,
+            cwl_command_str,
+            cwl.InlineJavascriptRequirement(),
+            cwl.NetworkAccess(networkAccess=True),
+            cwl.ResourceRequirement(
+                coresMin=cpu_requirement,
+                ramMin=memory_requirement,
+                outdirMin=outdir_requirement,
+            ),
+        ]
+
+    def get_outdir_requirement(
         self, outdir: Union[WDL.Expr.Get, WDL.Expr.Apply]
     ) -> int:
         """Produce the memory requirement for the output directory from WDL runtime disks."""
         # This is yet to be implemented. After Feature Parity.
-        return 1
+        raise NotImplementedError()
 
     def get_input(self, input_name: str) -> str:
-        """Produce a consise, valid CWL expr/param reference lookup string for a given input name."""
+        """Produce a concise, valid CWL expr/param reference lookup string for a given input name."""
         if valid_js_identifier.match(input_name):
             return f"inputs.{input_name}"
-        return f'inputs["{input_name}"]'  # pragma: no cover
+        return f'inputs["{input_name}"]'
```

Comment:
I think this is for coverage? We have a lot of other code that's not covered yet, and causes Codecov to fail (e.g. #125 ). So I thought we could remove these and do a review of coverage prior to the final release?